api/v1alpha1/api.go

```diff
@@ -61,12 +61,21 @@ type AIGatewayRouteList struct {
 }
 
 // AIGatewayRouteSpec details the AIGatewayRoute configuration.
+//
+// +kubebuilder:validation:XValidation:rule="!has(self.parentRefs) || !has(self.targetRefs) || size(self.targetRefs) == 0", message="targetRefs is deprecated, use parentRefs only"
```

Comment:
this will reject previous config, is that the goal ?

Reply:
No this won’t reject existing ones as they don’t have parentRefs set hence this is always true